message/batch.go

```diff
@@ -183,12 +183,12 @@ func (c *batchCodec) Encode(msg Message, dest io.Writer, version primitive.Proto
 			return fmt.Errorf("cannot write BATCH query flags: %w", err)
 		}
 		if version.SupportsQueryFlag(primitive.QueryFlagSerialConsistency) && flags.Contains(primitive.QueryFlagSerialConsistency) {
-			if err = primitive.WriteShort(uint16(batch.SerialConsistency.Value), dest); err != nil {
+			if err = primitive.WriteShort(uint16(*batch.SerialConsistency), dest); err != nil {
```

Comment:
Does this change the behavior when `batch.SerialConsistency` is `nil` here? Or can it never be `nil` here?